BuildResidentialScheduleFile/resources/schedules.rb

```diff
@@ -1074,6 +1123,75 @@ def get_building_america_lighting_schedule(time_zone_utc_offset, latitude, longi
     return lighting_sch
   end
 
+  # TODO
+  #
+  # @param away_schedule [TODO] TODO
+  # @param hours_driven_per_year [TODO] TODO
+  # @return [TODO] TODO
+  def _get_ev_battery_schedule(away_schedule, hours_driven_per_year)
+    total_driving_minutes_per_year = (hours_driven_per_year * 60).ceil
+    expanded_away_schedule = away_schedule.flat_map { |status| [status] * 15 }
+    charging_schedule = []
+    discharging_schedule = []
+    driving_minutes_used = 0
+    chunk_counts = expanded_away_schedule.chunk(&:itself).map { |value, elements| [value, elements.size] }
+    total_away_minutes = chunk_counts.map { |value, size| value * size }.sum
+    extra_drive_minutes = 0 # accumulator for keeping track of extra driving minutes used due to ceil to upper integer
+    chunk_counts.each do |is_away, activity_minutes|
+      if is_away == 1
+        current_chunk_proportion = (1.0 * activity_minutes) / total_away_minutes
+
+        expected_driving_time = (total_driving_minutes_per_year * current_chunk_proportion - extra_drive_minutes).ceil
+        max_driving_time = [expected_driving_time, total_driving_minutes_per_year - driving_minutes_used].min
+
+        max_possible_driving_time = (activity_minutes * 0.8).ceil
+        actual_driving_time = [max_driving_time, max_possible_driving_time].min
+        extra_drive_minutes += actual_driving_time - total_driving_minutes_per_year * current_chunk_proportion
+
+        idle_time = activity_minutes - actual_driving_time
+        first_half_driving = (actual_driving_time / 2.0).ceil
+        second_half_driving = actual_driving_time - first_half_driving
+
+        discharging_schedule += [1] * first_half_driving  # Start driving
+        discharging_schedule += [0] * idle_time           # Idle in the middle
+        discharging_schedule += [1] * second_half_driving # End driving
+        charging_schedule += [0] * activity_minutes
+
+        driving_minutes_used += actual_driving_time
+      else
+        charging_schedule += [1] * activity_minutes
+        discharging_schedule += [0] * activity_minutes
+      end
+    end
+    if driving_minutes_used < total_driving_minutes_per_year
+      msg = "Insufficient away minutes (#{total_away_minutes}) for required driving minutes (#{hours_driven_per_year * 60})"
+      msg += "Only #{driving_minutes_used} minutes was used."
+      @runner.registerWarning(msg)
+    end
+
+    return charging_schedule, discharging_schedule
+  end
+
+  # TODO
+  #
+  # @param markov_chain_simulation_result [TODO] TODO
+  # @return [TODO] TODO
+  def fill_ev_battery_schedule(markov_chain_simulation_result)
+    if @hpxml_bldg.vehicles.to_a.empty?
+      return
+    end
```

Comment:
I'm a bit confused. I see that no `ev_battery_charging` and `ev_battery_charging` columns are created when there's no EV. This is different than other things (like clothes washers), where we create schedules even when the appliance is not present. Why?

It's particularly confusing that when you have no EV, you still get an `ev_occupant` (and `present_occupants`) columns. Why?

Finally, I ran this measure on the `base-battery-ev.xml` file and it's showing 7 occupants?! The HPXML file has 3 bedrooms and doesn't have the number of occupants specified. Where did 7 come from?

Reply:
Yeah, this is the part we need to decide one way or the other. For other appliance, we can just generate the schedule based on the markov chain. For EV it's not so straight forward since it needs annual driving hours as an input to generate the schedule but that's not available if there is no EV in the HPXML. So, currently, I am not generating the EV schedule. But, yes, this is inconsistent with what we do with other appliances. We do have that input from ResStock though (even if there is no EV) - it's just not available in the HPXML file. We need to find a solution here.